frontend/Auth/Login.tsx

```diff
@@ -1,9 +1,167 @@
-import { Button, Form } from "semantic-ui-react";
+/*
+[x] new frontend route /login. 
+[x] contain a card with inputs for email, password, and a submit button.
+
+[x] When the form submits, it should make a request to the backend to login, possibly sending the password as a base64 encoded string.
+
+[x] If the backend responds with a 200, it can be assumed that a cookie was set and the user is now logged in. At that time, the user should be redirected back to the home page.
+*/
+
+import React, { useState } from "react";
+import { useHistory, Link } from "react-router-dom";
+import { Button, Form, Card } from "semantic-ui-react";
+import { RouterProps } from "react-router";
+import { useMutation } from "react-query";
+import { flaxFetch } from "../Utils/flaxFetch";
+import { unary } from "lodash-es";
+
+export function Login(props: RouterProps) {
+  const [loginFormData, setLoginFormData] = useState<LoginFormData>({
+    username: "",
+    password: "",
+  });
+
+  const history = useHistory();
+
+  const submitMutation = useMutation<
+    LoginResultData,
+    Error,
+    React.FormEvent<HTMLFormElement>
+  >(
+    (evt) => {
+      evt.preventDefault();
+      let requestPromise = flaxFetch<LoginResultData>(`/login`, {
+        method: "POST",
+        body: {
+          username: loginFormData.username,
+          password: loginFormData.password,
+        },
+      });
+      return requestPromise;
+    },
+    {
+      onSuccess: async (data, variables, context) => {
+        history.push("/home");
+      },
+      onError: (error, variables, context) => {
+        console.log(error);
+      },
+    }
+  );
 
-export function Login(props) {
   return (
-    <Form method="post" action="/login">
-      <Button type="submit">Login</Button>
-    </Form>
+    <div className="flex flex-col items-center grid grid-cols-3 gap-r">
+      <div className="flex items-center"></div>
+
+      <div>
+        <h1 className="text-xl text-gray-500 place-self-start pt-40">
+          Sign in
+        </h1>
+        <br />
```

Comment:
Usually better to use margin-bottom rather than a br, as br is a different size in different browsers and resolutions, and also is an old and possibly deprecated feature of HTML from back when they try to get HTML to do the styling rather than CSS